### DISCUSSION TITLE
接收消息的语法应该怎么写？

### DISCUSSION THREAD

**Discussion by Wybxc** [GraiaProject/Application#96] [Q&A]
这是文档中的示例：
```python
@bcc.receiver("FriendMessage")
async def friend_message_listener(app: GraiaMiraiApplication, friend: Friend):
    await app.sendFriendMessage(friend, MessageChain.create([
        Plain("Hello, World!")
    ]))
```
我有几个问题：
1. 用于接收消息的函数一定需要是 `async` 的吗？
2. 这个函数需要有哪些参数？可以有哪些参数？
3. 问题2应该和 `Dispatcher` 有关，但是 `Dispatcher` 的文档我看不懂……

**Comment by xian18** [comment]
1.不一定要是`async`的的。这取决于你这个函数是不是异步的。
可以看看社区提供的这个。 [asyncio](https://graiax.xyz/docs/appendixs/asyncio-basic)
比如
```python
@bcc.receiver("FriendMessage")
def friend_message_listener():
    print("收到一条好友消息")
```

2.这个问题还真的跟DIspatcher有关。。这个函数可以没有任何参数（比如上面这个）。也可以有任何参数，只不过需要对应Dispatcher来提供参数。否则会导致Requirement Crash。
至于DIspathcer咋用。。我只有一个写的稀烂的代码。。 [code src](https://github.com/xlxyzzgzs/a-qqbot-based-graia/blob/4674bc91945a710320aed5b9e697c8e6d50081cf/func/AnalyzeAppXmlJson.py#L21)，其中的`customDIspather`是定义的一个Dispatcer类，提供我想要的功能。
